File: agents/pm.py
"""
Project Manager Agent  (Sonnet — structured planning, moderate complexity)

Input : BRD + Architecture Document
Output: Project Plan dict
"""

from __future__ import annotations
import json
from config import MODELS
from core.claude import call_claude, parse_json

_SYSTEM = """You are a senior Project Manager AI with Agile expertise.

INPUT : BRD + Architecture Document JSON
OUTPUT: Project Plan JSON — respond with ONLY valid JSON, no prose.

Required structure:
{
  "project_name": "<name>",
  "duration_weeks": <total weeks>,
  "methodology": "<Scrum|Kanban|SAFe|etc>",
  "phases": [
    {"name": "<name>", "duration_weeks": <weeks>,
     "description": "<what happens>",
     "deliverables": ["<deliverable>"],
     "dependencies": ["<dependency>"]}
  ],
  "team_structure": [
    {"role": "<role>", "count": <number>,
     "responsibilities": ["<responsibility>"]}
  ],
  "milestones": [
    {"name": "<name>", "week": <week number>,
     "criteria": ["<success criterion>"]}
  ],
  "risks": [
    {"id": "R-001", "description": "<risk>",
     "probability": "LOW|MEDIUM|HIGH",
     "impact": "LOW|MEDIUM|HIGH",
     "mitigation": "<strategy>", "owner": "<role>"}
  ],
  "communication_plan": "<meetings, reporting cadence, tools>"
}"""


async def pm_generate(brd: dict, architecture: dict) -> dict:
    msg = (
        "Create a project plan based on:\n\n"
        f"BRD:\n{json.dumps(brd, indent=2)}\n\n"
        f"ARCHITECTURE:\n{json.dumps(architecture, indent=2)}"
    )
    raw = await call_claude(MODELS["pm"], _SYSTEM, [{"role": "user", "content": msg}])
    return parse_json(raw)

File: src/agents/pm.ts
/**
 * Project Manager Agent
 *
 * Consumes BRD + Architecture Document (JSON) → Produces Project Plan (JSON).
 * No natural language between agents.
 */

import { callClaude, parseJSON } from '../utils/claude.js';
import type { BRD, ArchitectureDocument, ProjectPlan } from '../types/index.js';

const SYSTEM_PROMPT = `You are a senior Project Manager AI with expertise in Agile delivery.

INPUT: A BRD and Architecture Document in JSON format.
OUTPUT: A detailed Project Plan in JSON format.

You MUST respond with ONLY valid JSON — no prose, no markdown, no code blocks.

Required output structure:
{
  "project_name": "<project name>",
  "duration_weeks": <total number of weeks>,
  "methodology": "Scrum|Kanban|SAFe|etc",
  "phases": [
    {
      "name": "<phase name>",
      "duration_weeks": <weeks>,
      "description": "<what happens in this phase>",
      "deliverables": ["<deliverable 1>"],
      "dependencies": ["<dependency>"]
    }
  ],
  "team_structure": [
    {
      "role": "<role name>",
      "count": <number>,
      "responsibilities": ["<responsibility 1>"]
    }
  ],
  "milestones": [
    {
      "name": "<milestone name>",
      "week": <week number>,
      "criteria": ["<success criterion 1>"]
    }
  ],
  "risks": [
    {
      "id": "R-001",
      "description": "<risk description>",
      "probability": "LOW|MEDIUM|HIGH",
      "impact": "LOW|MEDIUM|HIGH",
      "mitigation": "<mitigation strategy>",
      "owner": "<role responsible>"
    }
  ],
  "communication_plan": "<how the team communicates, meetings, reporting cadence>"
}

Planning guidelines:
- Be realistic about timelines — account for testing, review, and buffer
- Identify the critical path and dependencies
- Include discovery/design phases before development
- Team size should match project complexity
- Identify top 5-7 risks with concrete mitigations`;

export async function pmGenerateProjectPlan(
  brd: BRD,
  architecture: ArchitectureDocument
): Promise<ProjectPlan> {
  const userMessage = `Please create a project plan based on the following BRD and Architecture Document:

BRD:
${JSON.stringify(brd, null, 2)}

ARCHITECTURE DOCUMENT:
${JSON.stringify(architecture, null, 2)}

Produce a complete Project Plan following the specified JSON format.`;

  const raw = await callClaude(SYSTEM_PROMPT, [{ role: 'user', content: userMessage }], 16000);
  return parseJSON<ProjectPlan>(raw);
}
